### ISSUE TITLE
Fix 4chan's "/adv/ - Advice" board

### ISSUE THREAD

**Issue by OverMighty** [uBlockOrigin/uAssets#8900]
### Prerequisites

- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- [x] I tried to reproduce the issue when...
    - [ ] uBlock Origin is the only extension
    - [x] uBlock Origin with default lists/settings
    - [ ] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin

### URL(s) where the issue occurs

- `https://boards.4channel.org/adv/`
- `https://boards.4channel.org/adv/thread/*`

### Describe the issue

 

The following features are broken on 4chan's "/adv/ - Advice" board:

- posting replies using the quick reply panel that shows up when clicking on a post's ID (posting replies with the [Post a Reply] button at the top of the thread's page works)
- updating threads using the [Update] buttons
- the automatic, periodic purge of dead threads from the local storage used by the "Thread hiding" feature of 4chan's native extension (not a browser extension, just additional JavaScript code) (see https://github.com/4chan/4chan-JS/blob/master/extension.js#L4601-L4659)

### Screenshot(s)

![image](https://user-images.githubusercontent.com/33205782/115018826-a4bdb800-9eb8-11eb-879c-9f1a6e4fe778.png)

### Versions

- Browser/version: Chromium/89.0.4389.114 (Official Build) Arch Linux (64-bit)
- uBlock Origin version: 1.34.1rc0

### Settings

Default. Clicked the "Reset to default settings..." button in the uBlock Origin dashboard then just clicked the "Update now" button in the "Filter lists" tab.

### Notes

I found how to fix this issue and will submit a pull request.

**Comment by krystian3w** [comment]
Possible reporduce with other subdomain than `boards.` (?), In most cases subdomain / `www.` is omitted.

**Comment by OverMighty** [comment]
I couldn't find a page on `www.4chan.org` and `www.4channel.org` that makes requests to `a.4cdn.org/*.json`, only pages on `boards.4chan.org` and `boards.4channel.org`. Should I still broaden the filters to all of `4chan.org` and `4channel.org`?

**Comment by uBlock-user** [comment]
@ryanbr

**Comment by ryanbr** [comment]
https://github.com/easylist/easylist/commit/cd74833bc2c71896425d52f7252d3a961f7d68a9  is enough

**Comment by OverMighty** [comment]
[easylist/easylist@cd74833](https://github.com/easylist/easylist/commit/cd74833bc2c71896425d52f7252d3a961f7d68a9) does not fix `sys.4channel.org/adv/post` getting blocked.

**Comment by ryanbr** [comment]
I clicked on `sys.4channel.org/adv/post` and it redirected me to https://boards.4channel.org/adv/ no issues here.

**Comment by OverMighty** [comment]
That's because "doc" requests to `sys.4channel.org/adv/post` aren't blocked, only XHRs. This is also why the [Post a Reply] buttons work: it's part of a `<form>` so clicking it triggers a "doc" request and not an XHR.

**Comment by uBlock-user** [comment]
@ryanbr Best to set `$domain` to `4channel.org` instead of `board.4channel.org` as to avoid any other blockages of this type.

**Comment by OverMighty** [comment]
When posting a reply using the [Post a Reply] button at the top of the thread page:

![image](https://user-images.githubusercontent.com/33205782/115037495-d0e43380-9ece-11eb-9a00-9eb2537a6782.png)

When posting a reply using the quick reply panel (as shown in the screenshot in my issue-opening comment):

![image](https://user-images.githubusercontent.com/33205782/115037629-f5d8a680-9ece-11eb-9a83-063d9ef520bf.png)

**Comment by OverMighty** [comment]
I found why the quick reply panel triggers an XHR instead of a "doc" request like other `<form>`s: JavaScript is used to listen to the `submit` event and replace the default form submit logic with JavaScript and XHR.

https://github.com/4chan/4chan-JS/blob/master/extension.js#L4128-L4134

https://github.com/4chan/4chan-JS/blob/master/extension.js#L4220-L4353

**Comment by OverMighty** [comment]
XHRs to `​https://sys.4channel.org/adv/post` are still blocked by EasyList due to the `.org/adv/` part, preventing the use of the quick reply panel.

Should I open a new issue on the EasyList repository?
